Description: Derived from upstream patch which fixes an issue in which client certificate authentication fails
Origin:      upstream, https://gitorious.org/gnutls/gnutls/commit/555766063e08fc675b88e06560f79456c4ba4f24?format=diff
Bug-Ubuntu:  https://bugs.launchpad.net/bugs/1095052
--- a/lib/ext_signature.c
+++ b/lib/ext_signature.c
@@ -127,7 +127,7 @@
 _gnutls_sign_algorithm_parse_data (gnutls_session_t session,
                                    const opaque * data, size_t data_size)
 {
-  int sig, i;
+  int sig, i, hash;
   sig_ext_st *priv;
   extension_priv_data_t epriv;
 
@@ -149,6 +149,10 @@
 
        _gnutls_debug_log ("EXT[SIGA]: rcvd signature algo (%d.%d) %s\n", aid.hash_algorithm, 
          aid.sign_algorithm, gnutls_sign_get_name(sig));
+
+      hash = _gnutls_sign_get_hash_algorithm(sig);
+      if (hash != GNUTLS_DIG_SHA1 && hash != GNUTLS_DIG_SHA256)
+        continue;
 
       if (sig != GNUTLS_SIGN_UNKNOWN)
         {
